src/java.base/share/classes/java/lang/invoke/LambdaForm.java

```diff
@@ -330,55 +330,69 @@ private Kind(String defaultLambdaName, String methodName) {
         }
     }
 
-    LambdaForm(int arity, Name[] names, int result) {
-        this(arity, names, result, /*forceInline=*/true, /*customized=*/null, Kind.GENERIC);
-    }
-    LambdaForm(int arity, Name[] names, int result, Kind kind) {
-        this(arity, names, result, /*forceInline=*/true, /*customized=*/null, kind);
-    }
-    LambdaForm(int arity, Name[] names, int result, boolean forceInline, MethodHandle customized) {
-        this(arity, names, result, forceInline, customized, Kind.GENERIC);
-    }
-    LambdaForm(int arity, Name[] names, int result, boolean forceInline, MethodHandle customized, Kind kind) {
-        assert(namesOK(arity, names));
+    // private version that doesn't do checks or defensive copies
+    private LambdaForm(int arity, int result, boolean forceInline, MethodHandle customized, Name[] names, Kind kind) {
         this.arity = arity;
-        this.result = fixResult(result, names);
-        this.names = names.clone();
+        this.result = result;
         this.forceInline = forceInline;
         this.customized = customized;
+        this.names = names;
         this.kind = kind;
-        int maxOutArity = normalize();
-        if (maxOutArity > MethodType.MAX_MH_INVOKER_ARITY) {
-            // Cannot use LF interpreter on very high arity expressions.
-            assert(maxOutArity <= MethodType.MAX_JVM_ARITY);
-            compileToBytecode();
+        this.vmentry = null;
+        this.isCompiled = false;
+    }
+
+    // root factory pre/post processing and calls simple cosntructor
+    private static LambdaForm create(int arity, Name[] names, int result, boolean forceInline, MethodHandle customized, Kind kind) {
+        names = names.clone();
```

Comment:
Probably not important - and not for this RFE - but looking through the calls it seems that for some uses of `create` cloning the input array is superfluous. Maybe there's a very slight efficiency gain to be had here by making cloning the responsibility of the caller.